File: android/src/androidTest/java/org/ligi/gobandroid_hd/uitest/TestComponent.java
package org.ligi.gobandroid_hd.uitest;

import dagger.Component;
import javax.inject.Singleton;
import org.ligi.gobandroid_hd.etc.AppComponent;
import org.ligi.gobandroid_hd.etc.AppModule;

@Singleton
@Component(modules = {AppModule.class})
public interface TestComponent extends AppComponent {

    void inject(TheGameInfoDialog theGameInfoDialog);

    void inject(TheTsumegoActivity theTsumegoActivity);

    void inject(TheGameScoringActivity theGameScoringActivity);

    void inject(TheReviewActivity theReviewActivity);


    void inject(TheGoGamePlayerActivity theGoGamePlayerActivity);

    void inject(TheEditGameActivity theEditGameActivity);

    void inject(TheUndoFunctionality theUndoFunctionality);
}

File: android/src/androidTest/java/org/ligi/gobandroid_hd/uitest/TestComponent.kt
package org.ligi.gobandroid_hd.uitest

import dagger.Component
import org.ligi.gobandroid_hd.etc.AppComponent
import org.ligi.gobandroid_hd.etc.AppModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface TestComponent : AppComponent {

    fun inject(theGameInfoDialog: TheGameInfoDialog)

    fun inject(theTsumegoActivity: TheTsumegoActivity)

    fun inject(theGameScoringActivity: TheGameScoringActivity)

    fun inject(theReviewActivity: TheReviewActivity)

    fun inject(theGoGamePlayerActivity: TheGoGamePlayerActivity)

    fun inject(theEditGameActivity: TheEditGameActivity)

    fun inject(theUndoFunctionality: TheUndoFunctionality)
}
